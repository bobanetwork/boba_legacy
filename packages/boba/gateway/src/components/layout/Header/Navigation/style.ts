import { NavLink } from 'react-router-dom'
import styled, { css } from 'styled-components'
import { mobile } from 'themes/screens'

export const StyledNav = styled.div`
  display: flex;
  gap: 24px;
  ${mobile(css`
    display: none;
  `)}
`

export const NavLinkItem = styled(NavLink)`
  font-size: 14px;
  line-height: 16px;
  font-weight: 500;
  text-transform: uppercase;
  cursor: pointer;
  text-decoration: none;
  color: ${({ theme }) =>
    theme.name === 'light'
      ? theme.colors['gray'][600]
      : theme.colors['gray'][100]};
  &:hover,
  &.active {
    color: ${({ theme }) =>
      theme.name === 'light'
        ? theme.colors['gray'][800]
        : theme.colors['green'][300]};
  }
`
