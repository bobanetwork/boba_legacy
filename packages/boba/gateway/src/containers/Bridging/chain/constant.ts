import { ElementType } from 'react'

import AvalancheIcon from 'components/icons/chain/L1/AvalancheIcon'
import BNBIcon from 'components/icons/chain/L1/BNBIcon'
import EthereumIcon from 'components/icons/chain/L1/EthereumIcon'

import BobaAvaxIcon from 'components/icons/chain/L2/BobaAvaxIcon'
import BobaBNBIcon from 'components/icons/chain/L2/BobaBNBIcon'
import BobaIcon from 'components/icons/chain/L2/BobaIcon'

type IconType = {
  L1: ElementType
  L2: ElementType
}

type NetworkIconsType = Record<string, IconType>

export const NETWORK_ICONS: NetworkIconsType = {
  ethereum: {
    L1: EthereumIcon,
    L2: BobaIcon,
  },
  bnb: {
    L1: BNBIcon,
    L2: BobaBNBIcon,
  },
  avax: {
    L1: AvalancheIcon,
    L2: BobaAvaxIcon,
  },
}
