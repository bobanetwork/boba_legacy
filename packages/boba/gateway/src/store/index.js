/*
Copyright 2021-present Boba Network.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

import { createStore , applyMiddleware } from 'redux'
import RootReducer from 'reducers'
import reduxThunk from 'redux-thunk'
import persistReducer from 'redux-persist/lib/persistReducer'
import storage from 'redux-persist/lib/storage'
import persistStore from 'redux-persist/lib/persistStore'

const initialState = {}

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['network']
}

const persistedReducer = persistReducer(persistConfig, RootReducer)

export const store = createStore(
  persistedReducer,
  initialState,
  applyMiddleware(reduxThunk)
)

export default store;

export const persistor = persistStore(store)
