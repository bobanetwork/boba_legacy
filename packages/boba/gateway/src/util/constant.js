require('dotenv').config()
export const POLL_INTERVAL = process.env.REACT_APP_POLL_INTERVAL

export const BRIDGE_TYPE = {
  FAST_BRIDGE: "FAST_BRIDGE",
  CLASSIC_BRIDGE: "CLASSIC_BRIDGE",
  MULTI_BRIDGE: "MULTI_BRIDGE", /// FIXME: remove me
}
