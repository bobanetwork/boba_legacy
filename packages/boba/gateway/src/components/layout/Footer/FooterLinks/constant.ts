import { ROUTES_PATH } from 'util/constant'

interface LinkProps {
  label: string
  path: string
  isNav?: boolean
}

export const FOOTERLINKS: Array<LinkProps> = [
  {
    label: 'FAQs',
    path: 'https://docs.boba.network/faq',
  },
  {
    label: 'Dev Tools',
    path: ROUTES_PATH.DEV_TOOLS,
    isNav: true,
  },
  {
    label: 'Bobascope',
    path: ROUTES_PATH.BOBASCOPE,
    isNav: true,
  },
]

export const FOOTERLINKS_RIGHT: Array<LinkProps> = [
  {
    label: 'Boba Network Website',
    path: 'https://boba.network',
  },
  {
    label: 'Terms of Use',
    path: 'https://boba.network/terms-of-use/',
  },
]
