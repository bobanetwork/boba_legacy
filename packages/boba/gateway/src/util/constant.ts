import { addMonths, Now, addYear, Dayjs } from 'util/dates'

/**************
 * Env Params *
 **************/
/* eslint-disable @typescript-eslint/no-var-requires */
require('dotenv').config()

type EnvType = string | number | null | undefined

export const POLL_INTERVAL: EnvType =
  process.env.REACT_APP_POLL_INTERVAL || 20000
export const GAS_POLL_INTERVAL: EnvType =
  process.env.REACT_APP_GAS_POLL_INTERVAL || 40000
export const GA4_MEASUREMENT_ID: EnvType =
  process.env.REACT_APP_GA4_MEASUREMENT_ID || null
export const APP_ENV: EnvType = process.env.REACT_APP_ENV || 'dev'
export const SENTRY_DSN: EnvType = process.env.REACT_APP_SENTRY_DSN || null
export const INFURA_ID: EnvType = process.env.REACT_APP_INFURA_ID
export const MAX_HEALTH_BLOCK_LAG: EnvType =
  process.env.REACT_APP_MAX_HEALTH_BLOCK_LAG
export const WALLET_VERSION: EnvType = process.env.REACT_APP_WALLET_VERSION
export const APP_STATUS: EnvType = process.env.REACT_APP_STATUS || 'normal'
export const SPEED_CHECK: EnvType = process.env.REACT_APP_SPEED_CHECK
export const TARGET_CHAIN_URL: EnvType = process.env.REACT_APP_TARGET_CHAIN_URL
// VE DAO FLAG
export const DISABLE_VE_DAO: EnvType = process.env.REACT_APP_DISABLE_VE_DAO
// WalletConnect FLAG
export const DISABLE_WALLETCONNECT: EnvType =
  process.env.REACT_APP_DISABLE_WALLETCONNECT
export const BOBABEAM_STATUS: EnvType =
  process.env.REACT_APP_BOBA_BEAM_STATUS || 0
export const BOBAOPERA_STATUS: EnvType =
  process.env.REACT_APP_BOBA_OPERA_STATUS || 0

type BridgeType = {
  FAST_BRIDGE: string
  CLASSIC_BRIDGE: string
  MULTI_BRIDGE: string
  MULTI_CHAIN_BRIDGE: string
}

export const BRIDGE_TYPE: BridgeType = {
  FAST_BRIDGE: 'FAST_BRIDGE',
  CLASSIC_BRIDGE: 'CLASSIC_BRIDGE',
  MULTI_BRIDGE: 'MULTI_BRIDGE', //fix me remove me
  MULTI_CHAIN_BRIDGE: 'MULTI_CHAIN_BRIDGE',
}

type ExpiryOptionType = {
  value: string | Dayjs
  label: string
}

export const EXPIRY_OPTIONS: ExpiryOptionType[] = [
  {
    value: addMonths(Now(), 3, 'YYYY-MM-DD'),
    label: '3 Months',
  },
  {
    value: addMonths(Now(), 7, 'YYYY-MM-DD'),
    label: '6 Months',
  },
  {
    value: addYear(1, 'YYYY-MM-DD'),
    label: '1 Year',
  },
]

export const ECOSYSTEM_CATEGORY: string[] = [
  'defi',
  'gamefi',
  'nft',
  'bridge',
  'wallet',
  'tool',
  'token',
]

/*********************
 * Routes Constants **
 *********************/

type RoutesPathType = {
  BRIDGE: string
  ECOSYSTEM: string
  WALLET: string
  HISTORY: string
  EARN: string
  LOCK: string
  STAKE: string
  HELP: string
  BOBASCOPE: string
  MONSTER: string
  VOTE_DAO: string
  DAO: string
  DEV_TOOLS: string
}

export const ROUTES_PATH: RoutesPathType = {
  BRIDGE: '/bridge',
  ECOSYSTEM: '/ecosystem',
  WALLET: '/wallet',
  HISTORY: '/history',
  EARN: '/earn',
  LOCK: '/lock',
  STAKE: '/stake',
  HELP: '/help',
  BOBASCOPE: '/bobascope',
  MONSTER: '/monster',
  VOTE_DAO: '/votedao',
  DAO: '/DAO',
  DEV_TOOLS: '/devtools',
}

export const PER_PAGE: number = 8

type Network = 'ethereum' | 'bnb' | 'avax' | 'fantom' | 'moonbeam' //we move this to global network type once we define this
type Page =
  | 'Bridge'
  | 'Ecosystem'
  | 'Wallet'
  | 'History'
  | 'Earn'
  | 'Stake'
  | 'DAO'
  | 'Monster'
type PagesByNetworkType = Record<Network, Page[]>

export const PAGES_BY_NETWORK: PagesByNetworkType = {
  ethereum: [
    'Bridge',
    'Ecosystem',
    'Wallet',
    'History',
    'Earn',
    'Stake',
    'DAO',
    'Monster',
  ],
  bnb: ['Bridge', 'Wallet', 'Earn', 'History'],
  avax: ['Bridge', 'Wallet', 'Earn', 'History'],
  fantom: ['Bridge', 'Wallet', 'Earn', 'History'],
  moonbeam: ['Bridge', 'Wallet', 'Earn', 'History'],
}

enum Layer {
  L1 = 'L1',
  L2 = 'L2',
}
export const LAYER: { [key: string]: Layer } = Layer

type NetworkIconType = 'ethereum' | 'boba'

type NetworkNameType = {
  L1: string
  L2: string
}

type DefaultNetworkType = {
  NAME: NetworkNameType
  ICON: Record<Layer, NetworkIconType>
}

export const DEFAULT_NETWORK: DefaultNetworkType = {
  NAME: {
    L1: 'Ethereum',
    L2: 'Boba',
  },
  ICON: {
    L1: 'ethereum',
    L2: 'boba',
  },
}

export const MM_EXTENTION_URL: string =
  'https://chrome.google.com/webstore/detail/metamask/nkbihfbeogaeaoehlefnkodbefgpgknn?hl=en'

export const MIN_NATIVE_L1_BALANCE: number = 0.002

export const BANXA_URL: string = 'https://boba.banxa.com/?'
