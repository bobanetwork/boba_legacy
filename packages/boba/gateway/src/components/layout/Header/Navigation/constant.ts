import { ROUTES_PATH } from 'util/constant'
import { IMenuItem } from './types'

export const MENU_LIST: Array<IMenuItem> = [
  {
    label: 'Bridge',
    path: ROUTES_PATH.BRIDGE,
  },
  {
    label: 'History',
    path: ROUTES_PATH.HISTORY,
  },
  {
    label: 'Earn',
    path: ROUTES_PATH.EARN,
  },
  {
    label: 'Stake',
    path: ROUTES_PATH.STAKE,
  },
  {
    label: 'Dao',
    path: ROUTES_PATH.DAO,
  },
]
