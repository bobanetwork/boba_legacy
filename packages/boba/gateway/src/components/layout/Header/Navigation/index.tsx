import React, { FC } from 'react'
import { StyledNav, NavLinkItem } from './style'
import { MENU_LIST } from './constant'
import { MenuProps } from './types'

/**
 *
 * Below pages will be available for all networks
 *
 * History
 * Ecosystem
 * Bridge
 * Earn
 *
 * Stake / Dao - Only available for eth Boba (Testnet / Mainnet)
 *
 * Filter is not required on menulist as we can force user to
 * connect to boba network on stake & dao page.
 *
 * @param isOpen - is the flag to open and show the menu in case of mobile view.
 *
 *
 * @returns
 */

const Navigation: FC<MenuProps> = ({ isOpen }) => {
  return (
    <StyledNav>
      {MENU_LIST.map((menu) => {
        return (
          <NavLinkItem
            key={menu.label}
            to={menu.path}
            activeclassname="active"
            // onClick={() => isMobile ? setOpen(false) : null}
          >
            {menu.label}
          </NavLinkItem>
        )
      })}
    </StyledNav>
  )
}

export default Navigation
