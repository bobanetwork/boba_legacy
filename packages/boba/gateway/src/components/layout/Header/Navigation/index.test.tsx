import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import CustomThemeProvider from 'themes'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import Navigation from '.'
import store from 'store'
import { MENU_LIST } from './constant'

const renderHeaderMenu = () => {
  return render(
    <MemoryRouter>
      <Provider store={store}>
        <CustomThemeProvider>
          <Navigation />
        </CustomThemeProvider>
      </Provider>
    </MemoryRouter>
  )
}

describe('Layout => Header => Menu', () => {
  test('should match the snapshot', () => {
    const { asFragment } = renderHeaderMenu()
    expect(asFragment()).toMatchSnapshot()
  })

  test('should have menu length of 6 with expected labels & paths', () => {
    renderHeaderMenu()
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(5)
    links.forEach((link, index) => {
      expect(link).toHaveTextContent(MENU_LIST[index].label)
      expect(link).toHaveAttribute('href', MENU_LIST[index].path)
    })
  })

  test('should change the location on clicking menu', () => {
    renderHeaderMenu()
    const links = screen.getAllByRole('link')
    links.forEach((link) => {
      fireEvent.click(link)
      expect(link).toHaveClass('active')
    })
  })
})
