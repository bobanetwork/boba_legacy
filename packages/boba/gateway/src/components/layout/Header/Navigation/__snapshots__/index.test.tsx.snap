// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout => Header => Menu should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bgqQcB bHAysJ"
  >
    <a
      activeclassname="active"
      class="sc-gTRrQi fGVotN"
      href="/bridge"
    >
      Bridge
    </a>
    <a
      activeclassname="active"
      class="sc-gTRrQi fGVotN"
      href="/history"
    >
      History
    </a>
    <a
      activeclassname="active"
      class="sc-gTRrQi fGVotN"
      href="/earn"
    >
      Earn
    </a>
    <a
      activeclassname="active"
      class="sc-gTRrQi fGVotN"
      href="/stake"
    >
      Stake
    </a>
    <a
      activeclassname="active"
      class="sc-gTRrQi fGVotN"
      href="/DAO"
    >
      Dao
    </a>
  </div>
</DocumentFragment>
`;
